src/MSBuild/XMake.cs

```diff
@@ -2614,6 +2614,48 @@ private static void StartLocalNode(CommandLineSwitches commandLineSwitches)
                     OutOfProcTaskHostNode node = new OutOfProcTaskHostNode();
                     shutdownReason = node.Run(out nodeException);
                 }
+                else if (nodeModeNumber == 8)
+                {
+                    // If FEATURE_NODE_REUSE is OFF, just validates that the switch is OK, and always returns False
+                    bool nodeReuse = ProcessNodeReuseSwitch(commandLineSwitches[CommandLineSwitches.ParameterizedSwitch.NodeReuse]);
+                    string[] lowPriorityInput = commandLineSwitches[CommandLineSwitches.ParameterizedSwitch.LowPriority];
+                    bool lowpriority = lowPriorityInput.Length > 0 && lowPriorityInput[0].Equals("true");
+
+                    // Since build function has to reuse code from *this* class and OutOfProcServerNode is in different assembly
+                    // we have to pass down xmake build invocation to avoid circular dependency
+                    Func<string, (int exitCode, string exitType)> buildFunction = (commandLine) =>
+                    {
+                        int exitCode;
+                        ExitType exitType;
+
+                        if (!s_initialized)
+                        {
+                            exitType = ExitType.InitializationError;
+                        }
+                        else
+                        {
+                            exitType = Execute(
+#if FEATURE_GET_COMMANDLINE
+                                    commandLine
+#else
+                                    QuotingUtilities.SplitUnquoted(commandLine).ToArray()
+#endif
+                                );
+                            exitCode = exitType == ExitType.Success ? 0 : 1;
+                        }
+                        exitCode = exitType == ExitType.Success ? 0 : 1;
+
+                        return (exitCode, exitType.ToString());
+                    };
+
+                    // commandLineSwitches[CommandLineSwitches.ParameterizedSwitch.ConsoleLoggerParameters]
+
+                    OutOfProcServerNode node = new(buildFunction);
+
+                    shutdownReason = node.Run(nodeReuse, lowpriority, out nodeException);
```

Comment:
Concerning `lowpriority` here. Is that good that if we start two builds with different priorities it should start two different server processes that would accept requests according to priority too? Could we instead not put priority in the handshake and switch the priority of the process according to the value we get, so we have one server process for both high priority and low priority?

Reply:
This would require more work than just changing the priority of the server process. Not only would it have to propagate to worker/taskhost nodes, but other child processes that inherited the priority from its MSBuild parent would have to be tweaked as well (*"that"* because we don't know if those processes inherited their priority from MSBuild or if they set it explicitly).

So probably doable, albeit with subtle issues, but probably not worth it.

Reply:
I see, yes, if it is more subtle than changing the priority right away, than it might be not worth it.

Reply:
Based on our discussion, I removed low prio and node reuse part of the hash.